messaging/src/main/java/org/axonframework/eventhandling/TrackingEventProcessor.java

```diff
@@ -338,6 +341,7 @@ private void processBatch(Segment segment, BlockingStream<TrackedEventMessage<?>
                     if (canHandle(trackedEventMessage, processingSegments)) {
                         batch.add(trackedEventMessage);
                     } else {
+                        eventStream.reportIgnored(trackedEventMessage);
```

Comment:
I think this comment holds for all the `canHandle(EventMessage<?>, Segment)` calls in this method.

So, we'd need a `canHandle()` method that disregards the segment, as otherwise you could always end up in the situation where you report a message as ignored whilst another segment might take care of it.
Or, do we have another idea how to tackle this?

Reply:
Additionally, I would expect test cases for these changes in the `TrackingEventProcessor`.